src/router/RedirectAuth.jsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Navigate, useLocation } from 'react-router-dom';
+import { useAuth } from '../context';
+
+export const RedirectAuth = ({ children }) => {
+  const {
+    authState: { token },
+  } = useAuth();
+
+  const location = useLocation();
+
+  return token ? (
+    <Navigate to={'/'} state={{ from: location }} replace />
+  ) : (
+    children
```

Comment:
as you are redirecting user to their route before they landed on Login/Signup page
consider using Outlet instead

Reply:
Replaced with `Outlet`, Thanks for the suggestion.